pkg/kubelet/remote/remote_runtime.go

```diff
@@ -500,20 +471,13 @@ func (r *RemoteRuntimeService) ContainerStats(containerID string) (*runtimeapi.C
 	resp, err := r.runtimeClient.ContainerStats(ctx, &runtimeapi.ContainerStatsRequest{
 		ContainerId: containerID,
 	})
-	r.cleanupErrorTimeouts()
-	r.errorMapLock.Lock()
-	defer r.errorMapLock.Unlock()
 	if err != nil {
-		lastMsg, ok := r.lastError[containerID]
-		if !ok || err.Error() != lastMsg || time.Since(r.errorPrinted[containerID]) >= identicalErrorDelay {
+		if r.logReduction.ShouldMessageBePrinted(err.Error(), containerID) {
 			klog.Errorf("ContainerStatus %q from runtime service failed: %v", containerID, err)
-			r.errorPrinted[containerID] = time.Now()
-			r.lastError[containerID] = err.Error()
 		}
 		return nil, err
 	}
-	delete(r.lastError, containerID)
-	delete(r.errorPrinted, containerID)
+	r.logReduction.ClearID(containerID)
```

Comment:
this is cleaner than the past logic, should this be in a defer further up?

Reply:
No.  Only if there is no error then we want to clear out the logReduction record.  If there was an error, we need to hold onto it.